owncloudApp/src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.kt

```diff
@@ -0,0 +1,467 @@
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        progressController = TransferProgressController(activity as ComponentsGetter?).also {
+            it.setProgressBar(view.findViewById<ProgressBar>(R.id.fdProgressBar))
+        }
+
+        updateFileDetails(false, false)
+
+        view.findViewById<View>(R.id.fdCancelBtn).setOnClickListener {
+            (mContainerActivity as FileDisplayActivity).cancelTransference(file)
+            fileDetailsViewModel.cancelCurrentDownload(file)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putParcelable(FileActivity.EXTRA_FILE, file)
+        outState.putParcelable(FileActivity.EXTRA_ACCOUNT, account)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        progressController?.startListeningProgressFor(file, account)
+        startObservingProgressForDownload()
+    }
+
+    override fun onStop() {
+        progressController?.stopListeningProgressFor(file, account)
+        stopObservingWorkers()
+        super.onStop()
+    }
+
+    private fun startObservingProgressForDownload() {
+        val safeAccount = account ?: return
+
+        fileDetailsViewModel.startListeningToDownloadsFromAccountAndFile(file = file, account = safeAccount)
+        fileDetailsViewModel.pendingDownloads.observe(this) { }
+        fileDetailsViewModel.ongoingDownload.observeWorkerTillItFinishes(
+            owner = this,
+            onWorkEnqueued = { handleDownloadEnqueued() },
+            onWorkRunning = { progress -> handleRunningDownload(progress) },
+            onWorkSucceeded = { handleSuccessDownload() },
+            onWorkFailed = { showMessageInSnackbar(getString(R.string.downloader_download_failed_ticker)) },
+            onWorkCancelled = { showMessageInSnackbar(getString(R.string.downloader_download_canceled_ticker)) },
+            removeObserverAfterNull = false
+        )
+    }
+
+    private fun handleDownloadEnqueued() {
+        requireView().findViewById<View>(R.id.fdProgressBlock).isVisible = true
+
+        requireView().findViewById<TextView>(R.id.fdProgressText).apply {
+            isVisible = true
+            text = String.format(getString(R.string.downloader_download_enqueued_ticker), file.fileName)
+        }
+    }
+
+    private fun handleRunningDownload(progress: Int) {
+        requireView().findViewById<ProgressBar>(R.id.fdProgressBar).apply {
+            isIndeterminate = false
+            setProgress(progress)
+            invalidate()
+        }
+    }
+
+    private fun handleSuccessDownload() {
+        updateFileDetails(forcedTransferring = false, refresh = false)
+
+        val fileDisplayActivity = activity as FileDisplayActivity
+        fileDetailsViewModel.navigateToPreviewOrOpenFile(fileDisplayActivity, file)
+    }
+
+    private fun stopObservingWorkers() {
```

Comment:
Are you sure this is not done automatically? 

From LiveData docs => `An observer added with a Lifecycle will be automatically removed if the corresponding Lifecycle moves to Lifecycle.State.DESTROYED state. This is especially useful for activities and fragments where they can safely observe LiveData and not worry about leaks: they will be instantly unsubscribed when they are destroyed.`

Anyway you could have a look at [StateFlow](https://developer.android.com/kotlin/flow/stateflow-and-sharedflow), it makes something similar.

Reply:
I was not sure about how it worked with configuration changes like rotation. But it seems that using viewLifecycleOwner instead of the fragment itself fixes the problem.